app/views/field_slips/_form.html.erb

```diff
@@ -0,0 +1,32 @@
+<%= form_with(model: field_slip) do |form| %>
+  <% if field_slip.errors.any? %>
+    <div style="color: red">
+      <h2><%= pluralize(field_slip.errors.count, "error") %> prohibited this field_slip from being saved:</h2>
+
+      <ul>
+        <% field_slip.errors.each do |error| %>
+          <li><%= error.full_message %></li>
+        <% end %>
+      </ul>
+    </div>
+  <% end %>
+
+  <div>
+    <%= form.label :observation_id, style: "display: block" %>
+    <%= form.number_field :observation_id %>
+  </div>
+
+  <div>
+    <%= form.label :project_id, style: "display: block" %>
+    <%= form.number_field :project_id %>
+  </div>
+
+  <div>
+    <%= form.label :code, style: "display: block" %>
+    <%= form.text_field :code %>
+  </div>
+
```

Comment:
The helper `text_field_with_label` should produce the Bootstrap markup here that we use elsewhere. 

We may need a new `number_field_with_label` helper to do that for number fields.

Reply:
Adopted.  Thanks!

Reply:
Glad it helped. I think you can drop the wrapping `<div>`s too. The Bootstrap form element classes will handle it.